src/Timeline.cpp

```diff
@@ -313,6 +420,25 @@ openshot::EffectBase* Timeline::GetClipEffect(const std::string& id)
 	return nullptr;
 }
 
+// Return the list of effects on all clips
+std::list<openshot::EffectBase*> Timeline::ClipEffects() const {
+
+	// Initialize the list
+	std::list<EffectBase*> ClipEffects;
+
+	// Loop through all clips
+	for (const auto& clip : clips) {
+		
+		// Get the clip's list of effects
+		std::list<EffectBase*> clipEffects = clip->Effects();
+
+		// Append the clip's effects to the list
+		ClipEffects.insert(ClipEffects.end(), clipEffects.begin(), clipEffects.end());
+	}
+
+	return ClipEffects;
```

Comment:
Please, _PLEASE_ don't use two distinct variables named `ClipEffects` and `clipEffects`, in a function named `ClipEffects`. That's just not OK. Neither variable should be named any permutation of "clip effects".

Call the output list `out` (or any better, but non-confusing, name) and the inner one just `effects`, or something.